### ISSUE TITLE
Can't pull in snapshot gradle plugin

### ISSUE THREAD

**Issue by 19lmyers** [realm/realm-kotlin#1654]
### How frequently does the bug occur?
Always

### Description
I can't get my multiplatform project (targets iOS and Android) to pick up the snapshot version of the gradle plugin or library. For some reason, it tries to download a snapshot that isn't present in the repository.

### Stacktrace & log output
```shell
A problem occurred configuring root project 'Taskify'.
> Could not resolve all files for configuration ':classpath'.
   > Could not find io.realm.kotlin:gradle-plugin:1.14.0-SNAPSHOT.
     Searched in the following locations:
       - https://oss.sonatype.org/content/repositories/snapshots/io/realm/kotlin/gradle-plugin/1.14.0-SNAPSHOT/maven-metadata.xml
       - https://oss.sonatype.org/content/repositories/snapshots/io/realm/kotlin/gradle-plugin/1.14.0-SNAPSHOT/gradle-plugin-1.14.0-20240131.082859-13.pom
```

### Can you reproduce the bug?
Always

### Reproduction Steps
libs.versions.toml:

```
realm-gradle-plugin = { module = "io.realm.kotlin:gradle-plugin", version.ref = "realm" }

realm-base = { module = "io.realm.kotlin:library-base", version.ref = "realm" }
realm-sync = { module = "io.realm.kotlin:library-sync", version.ref = "realm" }
```

root build.gradle:

```
buildscript {
    repositories {
        maven(url = "https://oss.sonatype.org/content/repositories/snapshots")
    }
    dependencies {
        classpath(libs.realm.gradle.plugin)
    }
}
```

module build.gradle:

```
plugins {
    id("io.realm.kotlin")
}

kotlin {
    androidTarget()

    iosX64()
    iosArm64()
    iosSimulatorArm64()

    sourceSets {
        commonMain.dependencies {
            implementation(libs.realm.base)
            implementation(libs.realm.sync)
        }
    }
}
```

### Version
1.14.0-SNAPSHOT

### What Atlas App Services are you using?
Atlas Device Sync

### Are you using encryption?
No

### Platform OS and version(s)
N/A

### Build environment
Android Studio version: Iguana 2023.2.1 RC 1
Android Build Tools version: 8.3.0-rc01
Gradle version: 8.5

**Comment by nhachicha** [comment]
Hi @19lmyers this is a [bug](https://issues.apache.org/jira/browse/MGPG-85) in the maven publish plugin we use in Github Action, I'm working on a fix. In the meantime I manually published `1.14.0-TEST-SNAPSHOT` (it doesn't have windows and Linux JVM targets) but it should cover your Android/iOS dependencies. Let me know if this works 👍

**Comment by 19lmyers** [comment]
It worked, thank you!
